goal_src/engine/camera/cam-update.gc

```diff
@@ -5,3 +5,735 @@
+     (plane-from-points arg1 s2-0 s1-0 sv-240 0)
+     (plane-from-points arg1 s0-0 s3-1 sv-240 1)
+     (plane-from-points arg1 s3-1 s2-0 sv-240 2)
+     (let ((t9-20 plane-from-points)
+           (t0-3 3)
+           )
+      (t9-20 arg1 s1-0 s0-0 sv-240 t0-3)
+      )
+     )
+    )
+   )
+  )
+
+;; definition for symbol *update-leaf-when-outside-bsp*, type symbol
+(define *update-leaf-when-outside-bsp* #t)
+
+;; definition for function update-visible
+(defun update-visible ((arg0 math-camera) (arg1 int) (arg2 uint) (arg3 uint))
+  (local-vars (a0-16 symbol) (s3-0 level-vis-info) (s4-0 level-vis-info))
+  (when (not *artist-fix-visible*)
+   (dotimes (s5-0 (-> *level* length))
+    (let ((v1-4 (-> *level* level s5-0)))
+     (when (= (-> v1-4 status) 'active)
+      (if
+       (and
+        (nonzero? (-> v1-4 bsp nodes))
+        (or *update-leaf-when-outside-bsp* (-> v1-4 inside-boxes?))
+        )
+       (bsp-camera-asm (-> v1-4 bsp) (-> arg0 trans))
+       )
+      )
+     )
+    )
+   (dotimes (gp-1 (-> *level* length))
+    (let ((s5-1 (-> *level* level gp-1)))
+     (when (= (-> s5-1 status) 'active)
+      (let ((a2-1 (/ (+ (-> s5-1 bsp visible-list-length) 15) 16)))
+       (set! (-> s5-1 all-visible?) (if (-> s5-1 all-visible?)
+                                     (-> s5-1 all-visible?)
+                                     #t
+                                     )
+        )
+       (cond
+        ((or (not (-> s5-1 force-all-visible?)) (-> s5-1 inside-boxes?))
+         (cond
+          ((or *artist-all-visible* (zero? (-> s5-1 bsp nodes)))
+           (quad-copy! (-> s5-1 vis-bits) (-> s5-1 bsp all-visible-list) a2-1)
+           )
+          ((begin
+            (set! s4-0 (-> s5-1 vis-info (-> s5-1 vis-self-index)))
+            (set! s3-0 (-> s5-1 vis-info (-> s5-1 vis-adj-index)))
+            (let ((v1-32 #f))
+             (set! a0-16 #f)
+             (when s4-0
+              (set!
+               v1-32
+               (nonzero? (logand (shl #x8000 16) (the-as int (-> s4-0 flags))))
+               )
+              (let ((a1-8 (-> s4-0 length)))
+               (set! arg3 (-> s4-0 from-bsp unk-data-2 0))
+               (if (< a1-8 arg3)
+                (set! v1-32 #f)
+                )
+               )
+              )
+             (when s3-0
+              (set!
+               a0-16
+               (nonzero? (logand (shl #x8000 16) (the-as int (-> s3-0 flags))))
+               )
+              (let ((a1-15 (-> s3-0 length)))
+               (set! arg3 (-> s3-0 from-bsp unk-data-2 0))
+               (if (< a1-15 arg3)
+                (set! a0-16 #f)
+                )
+               )
+              )
+             v1-32
+             )
+            )
+           (if (!= (-> s5-1 all-visible?) 'loading)
+            (set! (-> s5-1 all-visible?) #f)
+            )
+           (let* ((a0-19 s5-1)
+                  (t9-2 (method-of-object a0-19 dummy-16))
```

Comment:
This method should take more arguments so it ends up used directly in the `when` below